WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/ReportList.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftUI
+import Yosemite
+
+protocol ReportListPresentable {
+    var items: [ReportItem] { get }
+    var title: String { get }
+    var ctaTitle: String { get }
+    var onDismiss: () -> Void { get }
+}
+
+struct ReportItem: Identifiable {
+    let id = UUID()
+    let title: String
+    let subtitle: String
+    let iconUrl: String
+    let iconBase64: String?
+
+    var icon: Icon? {
+        if let base64String = iconBase64,
+           let imageData = Data(base64Encoded: base64String),
+           let image = UIImage(data: imageData) {
+            return .base64(image)
+        } else if let url = URL(string: iconUrl) {
+            return .remote(url)
+        }
+        return nil
+    }
```

Comment:
Can't we just inject the correct enum? I don't see why we would want to have those tho properties if one won't be used.

Reply:
Yep, makes total sense!
Done! ✅